### PR TITLE
Issue 20376 - @disable this(ref return scope Foo rhs) enables broken binaries (out-of-bounds access)

### PR DESCRIPTION
**Pull request by RazvanN7**
The conversion is handled by the copy constructor only when the return value is an lvalue and the object has a defined copy constructor.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Review by atilaneves — APPROVED** [review]

**Comment by RazvanN7** [comment]
I think that the buildkite failure is due to a version mismatch (dmd - stable branch, phobos/druntime - master branch). Is there any way I can fix this?

**Comment by wilzbach** [comment]
Well, the problem is that Buildkite doesn't detect master to stable changes and thus sets this:

```
BUILDKITE_PULL_REQUEST_BASE_BRANCH="master"
```

from: https://buildkite.com/dlang/dmd/builds/10208#703fe997-f4f1-40f8-bb8d-038b78a4628e
used in: https://github.com/dlang/ci/blob/master/buildkite/origin_target_branch.sh

tl;dr: a force-push will change fix the `BUILDKITE_PULL_REQUEST_BASE_BRANCH` variable.

Full discussion: https://github.com/buildkite/feedback/issues/306
